marlowe-dashboard-client/src/MainFrame/State.purs

```diff
@@ -225,7 +254,30 @@ handleAction (PlayAction (Play.AddNewWallet mTokenName)) = do
 handleAction (PlayAction playAction) = Play.handleAction playAction
 
 ------------------------------------------------------------
-mkNewWallet :: Nickname -> String -> RemoteData AjaxError PubKey -> Maybe WalletDetails
-mkNewWallet nickname contractId remoteDataPubKey = case remoteDataPubKey of
-  Success pubKey -> Just { nickname, contractId, pubKey, balance: Nothing }
-  _ -> Nothing
+emptyNewWalletDetails :: NewWalletDetails
+emptyNewWalletDetails =
+  { walletNicknameString: mempty
+  , contractInstanceIdString: mempty
+  , remoteDataWallet: NotAsked
```

Comment:
for later, I'd probably remove the remoteData prefix, the type already indicates it comes from the outside.

Reply:
I like it for clarity. Also removing it would give me naming problems with my lenses: I've got a `_wallet` lens on the `WalletDetails` (when we know we have one), and a `_remoteDataWallet` lens on the `NewWalletDetails`. Same for `_pubKey` / `_remoteDataPubKey`.